app/models/vaccination_record.rb

```
@@ -158,6 +159,8 @@ class VaccinationRecord < ApplicationRecord
               less_than_or_equal_to: -> { Time.current }
             }
 
+  validate :patient_attended_session
```

I don't think we need this in the `VaccinationRecord` model, only in the `DraftVaccinationRecord` model. The reason being that this means whenever we try and save a vaccination record (particular in bulk) we're now going to be running this validation which makes a request to the database which could have quite a significant performance impact.

Since the `DraftVaccinationRecord` route is the only way this issue can occur, I don't think it's needed in the model itself.

---

In that case, would it be better to also raise an error when the patient has already completed the session? I chose to allow this because otherwise the vaccination record would not be valid but if we are only running the check at :confirm step there is no need to allow the "completed" status.

---

I think we should still allow recording when the status is completed because a patient could return later that day. My understanding is this is a legitimate scenario:

- Patient refuses the vaccine
- Patient is marked as completed
- Patient returns later that day to get the vaccine